.github/workflows/main_unit_tests_mpich.yaml

```diff
@@ -12,15 +13,18 @@ jobs:
   main_unit_tests:
     runs-on: ubuntu-latest
     container:
-      image: ghcr.io/noaa-gfdl/pace_mpich:3.11.9
+      image: ghcr.io/noaa-gfdl/miniforge:mpich
     steps:
-        - name: Checkout Pace repository
-          uses: actions/checkout@v4
-          with:
+      - name: Checkout Pace repository
+        uses: actions/checkout@v4
+        with:
             submodules: 'recursive'
-        - name: setup env and run tests (test)
-          run: |
-            cp /home/scripts/setup_env.sh . && chmod +x setup_env.sh
-            cp /home/scripts/run_tests.sh . && chmod +x run_tests.sh
-            ./setup_env.sh
-            ./run_tests.sh
+      - name: setup env and generate input files
+        run: |
+          pip3 install --upgrade pip setuptools wheel
+          pip3 install -r requirements_dev.txt -c constraints.txt
+          cd tests/main && mkdir -p input && cd input
+          python3 ../../../examples/generate_eta_files.py
+          cd ../../../
+      - name:
```

Comment:
Do we want to name this step?

Reply:
Hi Oliver, I'm so sorry!  I decided to push in the reusable workflow changes to this PR, it might need your review again.  :innocent: